libcontainer/nsenter/nsexec.c

```diff
@@ -1225,8 +1226,24 @@ void nsexec(void)
 			 * was broken, so we'll just do it the long way anyway.
 			 */
 			write_log(DEBUG, "unshare remaining namespace (except cgroupns)");
-			if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) < 0)
-				bail("failed to unshare remaining namespaces (except cgroupns)");
+			/*
+			 * In centos 7.4/7.5/7.6, when other processes read the /proc/pid/status
+			 * file of the runc: [1: CHILD] process, and if the runc: [1: CHILD]
+			 * process happens to be in the unshare stage, the unshare syscall will
+			 * report an error EINVAL. This is because when other processes read
+			 * the status file, they will call the kernel function get_ task_ mm(),
+			 * which sets the decision condition task->mm->mm_users +1. In the
+			 * unshare syscall, mm does not meet the condition of mm<=1, and the
+			 * kernel throws an EINVAL error. For this, I think we can try again.
+			 */
+			for (; retries > 0; retries--) {
+				if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) >= 0)
```

Comment:
Doesn't matter much, but this should be `== 0` (the function returns `0` on success and `-1` on error.

Reply:
Done. Thanks for your suggestion. Why I use >=0 is the origin codes using if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) < 0). The Rigorous usage should be==0.

man 2 unshare:
RETURN VALUE
       On success, zero returned.  On failure, -1 is returned and errno is set
       to indicate the error.